ballerina/Module.md

```diff
@@ -92,6 +92,59 @@ sql:ExecutionResult result = check dbClient->execute("CREATE TABLE student(id IN
 //A value of the sql:ExecutionResult type is returned for 'result'. 
 ```
 
+### Parameterized Query
+
+The `ParameterizedQuery` is used to construct the SQL query to be executed by the client. 
+You can create a query with constant or dynamic input data as follows.
+
+*Query with constant values*
+
+```ballerina
+ParameterizedQuery query = `SELECT * FROM students WHERE 
+                            id < 10 AND age > 12`;
+```
+
+*Query with dynamic values*
+
+```ballerina
+int[] ids = [10, 50];
+int age = 12;
+ParameterizedQuery query = `SELECT * FROM students WHERE 
+                            id < ${ids[0]} AND age > ${age}`;
+```
+
+You can create the complex query by following the above samples. 
+But Ballerina has the following util functions that make it easier 
+to create a dynamic/constant complex query.
+
+* `queryConcat()`: This is used to create a parameterized query by concatenating sub-parameterized queries.
+  The sample below shows how to concatenate queries.
+    ```ballerina
+    int intType = 2147483647;
+    int bigIntType = 9223372036854774807;
+    int smallIntType = 32767;
+    ParameterizedQuery query = `INSERT INTO NumericTypes (int_type,
+                                bigint_type, smallint_type)`;
+    ParameterizedQuery query1 = ` VALUES(${intType}, ${bigIntType}, 
+                                ${smallIntType})`;
```

Comment:
Shall we have a sample with the where clause? Since that is the most common use case. We can have a basic select clause and conditions 1 and 2.